api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -17,17 +17,11 @@ namespace Microsoft.RedHatOpenShift;
  */
 
 /** HCP cluster resource */
-model HcpOpenShiftClusterResource
-  is TrackedResource<HcpOpenShiftClusterProperties> {
-  /** Name of HCP cluster */
-  @pattern("^[a-zA-Z][a-zA-Z0-9-]$")
-  @minLength(3)
-  @maxLength(54)
-  @key("hcpOpenShiftClusterName") // sets the alternative name for the name propery
-  @path
-  @segment("hcpOpenShiftClusters")
-  name: string;
-
+model HcpOpenShiftCluster is TrackedResource<HcpOpenShiftClusterProperties> {
+  ...ResourceNameParameter<
+    HcpOpenShiftCluster,
+    NamePattern = "^[a-zA-Z][-a-zA-Z0-9]{1,52}[a-zA-Z0-9]$"
```

Comment:
This is not correct because of the reasons mentioned in https://github.com/Azure/ARO-HCP/pull/1510/files#diff-4dec67f51651c81d6090e5ee3cf8021338af5ffd93e456d8c88eb585743aec9cR23.

Aside from that, if the intention is to simulate a RFC 1035 dns label minimum length the length check seems to also be wrong. The RFC 1035 allows something like "a", which we don't allow here. Here we are introducing a minimum of 3 characters. If we want to add this additional restriction of minimum 3 characters that's fine, but it is because of an additional restriction we decide, not of RFC 1035 dns label or CS.

The easiest would be separating the regex from the minimum and maximum lengths if possible

Reply:
> Here we are introducing a minimum of 3 characters. If we want to add this additional restriction of minimum 3 characters that's fine, but it is because of an additional restriction we decide, not of RFC 1035 dns label or CS.

FTR, 3 character minimum has been in place since the API spec was first drafted.  I'm not introducing it here, just coding it differently.  I don't know where it came from but I assume the BU has final say.